.github/workflows/rust.yml

```diff
@@ -9,7 +9,7 @@ on:
   merge_group:
 
 concurrency:
-  group: rust-${{ github.workflow }}-${{ github.ref }}
+  group: rust-${{ github.workflow }}-${{ github.event_name }}-${{ github.ref }}
```

Comment:
I don't understand how this impacts priority of anything anywhere

Reply:
`github.event_name` ensures push and merge_group do not share the same concurrency group.

Reply:
> `github.ref` for merge group and pushes to branches should already be different.

The refs for the merge queue and other workflows can be the same, if they are both "the branch that triggered the workflow run":
https://docs.github.com/en/enterprise-cloud@latest/actions/writing-workflows/choosing-what-your-workflow-does/accessing-contextual-information-about-workflow-runs#github-context

I'd support making this change to get better cancellation debugging, so we can see which events are causing the cancellations.

Reply:
You can already see the reason of the CI run from logs, I don't think event name is of much help here